server/src/main/java/org/opensearch/InvalidArgumentException.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch;
+
+import org.opensearch.common.io.stream.StreamInput;
+
+import java.io.IOException;
+
+/**
+ * Base exception for a missing action on a primary
+ *
+ * @opensearch.internal
+ */
+public class InvalidArgumentException extends OpenSearchException {
```

Comment:
Can the existing `org.opensearch.common.settings.SettingsException` class be used here and avoid creating a new one?

It is important that this case returns a BAD_REQUEST error code, so the following would need to be added to SettingsException to make that happen:

```
    @Override
    public RestStatus status() {
        return RestStatus.BAD_REQUEST;
    }
```

That might be okay because it looks like the existing usages of SettingsException would probably be better as a BAD_REQUEST, but I'm not 100% sure about that.